vcs/gitcmd/repo.go

```diff
@@ -962,6 +962,31 @@ func (r *Repository) Committers(opt vcs.CommittersOptions) ([]*vcs.Committer, er
 	return committers, nil
 }
 
+func (r *Repository) ListFiles(at vcs.CommitID) ([]string, error) {
+	r.editLock.RLock()
+	defer r.editLock.RUnlock()
+
+	if at == "" {
+		at = "HEAD"
+	}
+
+	cmd := exec.Command("git", "ls-files", "--with-tree", string(at))
```

Comment:
My thinking about `git ls-tree` and `git ls-files` is this:

- `git-ls-files` - Show information about files in the index and the working tree
  - This merges the file listing in the directory cache index with the actual working directory list, and shows different combinations of the two.
- `git-ls-tree` - List the contents of a tree object

As I understand it, `git-ls-files` is a superset of `git-ls-tree`. It includes everything tree does, but also augments it with files in current working directory.

go-vcs does not support current working directory, only individual commits.

You'd want to use it with these flags to get the results you want (no extra information):

`git ls-tree --full-tree -r --name-only <commitID>`